src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -1218,12 +1216,20 @@ void DoorLockServer::clearYearDayScheduleCommandHandler(chip::app::CommandHandle
     emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_SUCCESS);
 }
 
-bool DoorLockServer::HasFeature(chip::EndpointId endpointId, DoorLockFeature feature)
+bool DoorLockServer::HasFeatures(chip::EndpointId endpointId, uint32_t features)
+{
+    uint32_t featureMap = 0;
+    bool success        = GetAttribute(endpointId, Attributes::FeatureMap::Id, Attributes::FeatureMap::Get, featureMap);
```

Comment:
```
DoorLockFeature featureMap;
Attributes::FeatureMap::Get(endpoint, &featureMap);
```

to be typesafe.

Reply:
`GetAttribute` already logs the attribute access failure.